pgextwlist.c

```diff
@@ -415,14 +425,61 @@ call_ProcessUtility(PROCESS_UTILITY_PROTO_ARGS,
 						   | SECURITY_RESTRICTED_OPERATION);
 
 	if (action)
-		call_extension_scripts(name, schema, action,
-							   "before", old_version, new_version);
+	{
+		/* "drop extension" can list several extensions, walk them here */
+		if (!strcmp(action, "drop"))
+		{
+			Node   *parsetree = pstmt->utilityStmt;
+			ListCell *lc;
+			char   *name = NULL;
+
+			foreach(lc, ((DropStmt *)parsetree)->objects)
+			{
+				List *objname = lfirst(lc);
+#if PG_MAJOR_VERSION < 1000
+				name = strVal(linitial(objname));
+#elif PG_MAJOR_VERSION < 1500
+				name = strVal((Value *) objname);
+#else
+				name = strVal(castNode(String, objname));
+#endif
+				call_extension_scripts(name, schema, action,
+									   "before", old_version, new_version);
+			}
+		}
+		else
+			call_extension_scripts(name, schema, action,
+								   "before", old_version, new_version);
+	}
 
 	call_RawProcessUtility(PROCESS_UTILITY_ARGS);
 
 	if (action)
-		call_extension_scripts(name, schema, action,
-							   "after", old_version, new_version);
+	{
+		if (!strcmp(action, "drop"))
```

Comment:
I prefer using `if (strcmp(action, "drop") == 0)` and keep the ! operator for things that convey a sense of being false, or unsuccessful.